staging/src/k8s.io/apimachinery/pkg/util/httpstream/spdy/roundtripper.go

```diff
@@ -73,11 +80,33 @@ func NewSpdyRoundTripper(tlsConfig *tls.Config) *SpdyRoundTripper {
 	return &SpdyRoundTripper{tlsConfig: tlsConfig}
 }
 
-// implements pkg/util/net.TLSClientConfigHolder for proper TLS checking during proxying with a spdy roundtripper
+// TLSClientConfig implements pkg/util/net.TLSClientConfigHolder for proper TLS checking during
+// proxying with a spdy roundtripper.
 func (s *SpdyRoundTripper) TLSClientConfig() *tls.Config {
 	return s.tlsConfig
 }
 
+// SendRequest implements k8s.io/apimachinery/pkg/util/net.RequestSender.
+func (s *SpdyRoundTripper) SendRequest(method string, location *url.URL, header http.Header, body io.Reader) (net.Conn, error) {
```

Comment:
This method still looks almost identical to `UpgradeAwareProxyHandler.connectBackend`. Maybe abstracting at the dial method would be better?

```go
type Dialer interface {
    Dial(*http.Request) (net.Conn, error)
}
```